qiskit_experiments/autodocs/analysis_docs.py

```diff
@@ -0,0 +1,158 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Documentation for analysis class.
+"""
+import os
+import re
+from typing import Optional, Dict, List, Type
+
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Options
+
+from .descriptions import OptionsField, Reference, CurveFitParameter
+from .writer import _DocstringWriter, _CurveFitDocstringWriter, _DocstringMaker
+
+
+class StandardAnalysisDocstring(_DocstringMaker):
+    """A facade class to write standard analysis docstring."""
+
+    # pylint: disable=arguments-differ
+    @classmethod
+    def make_docstring(
+        cls,
+        default_options: Dict[str, OptionsField],
+        overview: Optional[str] = None,
+        example: Optional[str] = None,
+        references: Optional[List[Reference]] = None,
+        note: Optional[str] = None,
+        warning: Optional[str] = None,
+        tutorial: Optional[str] = None,
+    ) -> str:
+        try:
+            writer = _DocstringWriter()
+            if warning:
+                writer.write_warning(warning)
+            if overview:
+                writer.write_section(overview, "Overview")
+            if example:
+                writer.write_example(example)
+            if not isinstance(default_options, Options):
+                writer.write_lines(
+                    "Default class options. These options are automatically set "
+                    "when :py:meth:`run` method is called."
+                )
+                writer.write_options_as_sections(default_options, "Default Options")
+            else:
+                writer.write_lines(
+                    "Documentation for default options are not provided for this class."
+                )
+            if references:
+                writer.write_references(references)
+            if note:
+                writer.write_note(note)
+            if tutorial:
+                writer.write_tutorial_link(tutorial)
+        except Exception as ex:
+            raise QiskitError(f"Auto docstring generation failed with the error: {ex}") from ex
+        return writer.docstring
+
+
+class CurveAnalysisDocstring(_DocstringMaker):
```

Comment:
I understand your point, but we cannot make this subclass. Because order of calling writer method is important to standardize docstring section ordering. Although we can make standard one a superclass of this, we need to write full `make_docstring` method for this and there is almost no benefit from doing so.